cli/cache/check.rs

```diff
@@ -14,21 +13,24 @@ use super::common::run_sqlite_pragma;
 ///
 /// This simply stores a hash of the inputs of each successful type check
 /// and only clears them out when changing CLI versions.
-pub struct TypeCheckCache {
-  conn: Connection,
-}
+pub struct TypeCheckCache(Option<Connection>);
 
 impl TypeCheckCache {
-  pub fn new(db_file_path: &Path) -> Result<Self, AnyError> {
-    let conn = Connection::open(db_file_path).with_context(|| {
-      format!(
-        concat!(
-          "Error opening type checking cache at {} -- ",
-          "Perhaps it's corrupt. Maybe try deleting it."
-        ),
-        db_file_path.display()
-      )
-    })?;
+  pub fn new(db_file_path: &Path) -> Self {
+    match Self::try_new(db_file_path) {
+      Ok(cache) => cache,
+      Err(err) => {
+        log::debug!("Creating the type checking cache failed.\n{:#}", err);
```

Comment:
suggestion: warn that type checking is disabled if it was specified as enabled somewhere?  Clearly we are changing the behaviour and we shouldn't do that silently.

Reply:
Type checking isn't disabled in this case. When this fails it will always type check instead of potentially having a performance improvement and skipping type checking. So it's not really a behaviour change, but just a savings that someone won't have when it fails. Perhaps it would still be worthwhile to warn?

(I'm just going to update this to try again to create the db when it fails to open)

Reply:
Well it does warn, right?  The title just confused me in the sense I stopped reading the title I skipped the "cache" word. Any warning that tells them any action to take makes sense to me, but it is internal "performance" versus what I assumed what a change in behaviour.

Reply:
Currently it won't warn and will just show this message only when debug logging is enabled. I was thinking we could tell people to run with debug logging in case they find the type checking cache never works for them so we could get more details. I'm not sure if this will be a widespread problem.